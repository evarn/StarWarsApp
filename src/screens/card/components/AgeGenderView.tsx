import React from 'react';
import {StyleSheet, View, Text} from 'react-native';
import Fonts from '../../../constants/fonts';
import Colors from '../../../constants/colors';
import {getCheckFunction} from '../../../features/getCheckFunction';
import Strings from './../../../constants/strings';

interface IAgeGenderView {
  birth_year?: string | undefined;
  gender?: string | undefined;
}
const AgeGenderView = ({birth_year, gender}: IAgeGenderView) => {
  return (
    <View style={styles.mainContainer}>
      {getCheckFunction(birth_year) && (
        <View style={[styles.viewChar, {backgroundColor: Colors.BLUE_1}]}>
          <Text style={styles.textChar}>{birth_year} </Text>
        </View>
      )}

      {getCheckFunction(gender) && (
        <View
          style={[
            styles.viewChar,
            {
              backgroundColor:
                gender === Strings.INFO_MALE
                  ? Colors.GREEN_1
                  : gender === Strings.INFO_FEMALE
                  ? Colors.VIOLET_1
                  : Colors.YELLOW_1,
            },
          ]}>
          <Text style={styles.textChar}>{gender} </Text>
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  mainContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
  viewChar: {
    marginHorizontal: 4,
    padding: 8,
    borderRadius: 11,
  },
  textChar: {
    fontFamily: Fonts.SECONDARY,
    fontStyle: 'normal',
    fontWeight: '400',
    fontSize: 12,
    lineHeight: 14,
    textAlign: 'center',
    color: Colors.BLACK_2,
  },
});

export default AgeGenderView;
