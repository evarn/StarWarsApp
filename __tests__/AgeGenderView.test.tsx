import 'react-native';
import React from 'react';
import {render, cleanup} from '@testing-library/react-native';
import AgeGenderView from '../src/screens/card/components/AgeGenderView';
import Colors from '../src/constants/colors';
import {toHaveStyle} from '@testing-library/jest-native';

afterEach(cleanup);

describe('AgeGenderView', () => {
  expect.extend({toHaveStyle});
  it('AgeGenderView snapshoot', () => {
    const {toJSON} = render(
      <AgeGenderView birth_year={'69BBY'} gender={'male'} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('AgeGenderView male green  background color', () => {
    const {getByTestId} = render(
      <AgeGenderView birth_year={'69BBY'} gender={'male'} />,
    );
    const foundBodyElement = getByTestId('AgeGenderViewGender');
    expect(foundBodyElement).toHaveStyle({backgroundColor: Colors.GREEN_1});
  });

  it('AgeGenderView female violet background color', () => {
    const {getByTestId} = render(
      <AgeGenderView birth_year={'69BBY'} gender={'female'} />,
    );
    const foundBodyElement = getByTestId('AgeGenderViewGender');
    expect(foundBodyElement).toHaveStyle({backgroundColor: Colors.VIOLET_1});
  });
  it('AgeGenderView other yellow background color', () => {
    const {getByTestId} = render(
      <AgeGenderView birth_year={'69BBY'} gender={'other'} />,
    );
    const foundBodyElement = getByTestId('AgeGenderViewGender');
    expect(foundBodyElement).toHaveStyle({backgroundColor: Colors.YELLOW_1});
  });
  it('AgeGenderView birth year', () => {
    const {getByTestId} = render(
      <AgeGenderView birth_year={'69BBY'} gender={'other'} />,
    );
    const foundBodyElement = getByTestId('AgeGenderViewYear');
    expect(foundBodyElement).toHaveStyle({backgroundColor: Colors.BLUE_1});
  });
});
