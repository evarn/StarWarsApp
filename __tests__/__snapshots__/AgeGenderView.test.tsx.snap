// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgeGenderView AgeGenderView snapshoot 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
>
  <View
    style={
      [
        {
          "borderRadius": 11,
          "marginHorizontal": 4,
          "padding": 8,
        },
        {
          "backgroundColor": "#07D6F2",
        },
      ]
    }
    testID="AgeGenderViewYear"
  >
    <Text
      style={
        {
          "color": "#212121",
          "fontFamily": "Roboto",
          "fontSize": 12,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 14,
          "textAlign": "center",
        }
      }
    >
      69BBY
       
    </Text>
  </View>
  <View
    style={
      [
        {
          "borderRadius": 11,
          "marginHorizontal": 4,
          "padding": 8,
        },
        {
          "backgroundColor": "#73D677",
        },
      ]
    }
    testID="AgeGenderViewGender"
  >
    <Text
      style={
        {
          "color": "#212121",
          "fontFamily": "Roboto",
          "fontSize": 12,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 14,
          "textAlign": "center",
        }
      }
    >
      male
       
    </Text>
  </View>
</View>
`;
